mimium-lang/src/compiler/bytecodegen.rs

```diff
@@ -452,8 +459,14 @@ impl ByteCodeGenerator {
                     Self::word_size_for_type(*ty),
                 ))
             }
-            mir::Instruction::PushStateOffset(v) => Some(VmInstruction::ShiftStatePos(*v as i16)),
-            mir::Instruction::PopStateOffset(v) => Some(VmInstruction::ShiftStatePos(-(*v as i16))),
+            mir::Instruction::PushStateOffset(v) => {
+                let state_size = Self::calc_state_size(v) as i16;
+                Some(VmInstruction::SetStatePos(state_size))
+            }
+            mir::Instruction::PopStateOffset(v) => {
+                let state_size = Self::calc_state_size(v) as i16;
+                Some(VmInstruction::SetStatePos(0))
+            }
```

Comment:
Now this pull request succeeded to get rid of errors by breaking the concept of push/pop. I'm not sure if this direction is right (especially because I don't understand how `make_delay()` works). However, I couldn't come up with a nice way to calculate the relative offset in `emit_fncall()`.

Maybe this is something that I should not mess up unless I figure out the higher plans... :thinking: